src/js/components/Drop/StyledDrop.js

```diff
@@ -27,6 +27,98 @@ const dropKeyFrames = keyframes`
   }
 `;
 
+const isObject = object => object != null && typeof object === 'object';
+
+const deepEqual = (object1, object2) => {
+  const keys1 = Object.keys(object1);
+  const keys2 = Object.keys(object2);
+
+  if (keys1.length !== keys2.length) {
+    return false;
+  }
+
+  for (let i = 0; i < keys1.length; i += 1) {
+    const val1 = object1[keys1[i]];
+    const val2 = object2[keys1[i]];
+    const areObjects = isObject(val1) && isObject(val2);
+    if (
+      (areObjects && !deepEqual(val1, val2)) ||
+      (!areObjects && val1 !== val2)
+    ) {
+      return false;
+    }
+  }
+  return true;
+};
+
+// The desired margin may be adjusted depending on drops alignment
+const marginStyle = (theme, align, data, responsive, marginProp) => {
+  const margin = theme.global.edgeSize[data] || data;
+  let customCSS;
+  let adjustedMargin;
+  // if user provides CSS string such as '50px 12px', apply that always
+  if (typeof margin === 'string') {
+    customCSS = margin.split(' ').length > 1;
+  }
+  if (
+    theme.global.drop.intelligentMargin === true &&
+    !customCSS &&
+    typeof margin === 'string'
+  ) {
+    if (deepEqual(align, { top: 'bottom' })) {
```

Comment:
There's a lot of duplication in these if/else statements. How about something more like:
```
let adjustedMargin = {};
if (align.top === 'bottom') adjustedMargin.top = margin;
else if (align.bottom === 'top') adjustedMargin.bottom = margin;
if (align.right === 'left') adjustedMargin.left: `-${margin}`;
else if (align.left === 'right') adjustedMargin.left = margin;
if (!Object.keys(adjustedMargin)) adjustedMargin = 'none';
```